management/hostedservice/client.go

```diff
@@ -117,6 +117,19 @@ func (self HostedServiceClient) GetHostedService(name string) (HostedService, er
 	return hostedService, nil
 }
 
+func (self HostedServiceClient) ListHostedServices() ([]HostedService, error) {
+	var response listHostedServiceResponse
+
+	data, err := self.client.SendAzureGetRequest(azureHostedServiceListURL)
+	if err != nil {
+		return response.HostedServices, err
+	}
+
+	err = xml.Unmarshal(data, &response)
+
+	return response.HostedServices, err
```

Comment:
So when you do this:
```go
ss, _ = hsclient.ListHostedServices()
for _, s := range ss {
  // ... do something interesting with s
}
```
Won't that break if you change it form `type ListHostedServiceResponse []HostedService` to `type ListHostedServiceResponse struct`...

So then we'd want `ListHostedServiceResponse ` to be a `struct` and have the caller always dig into the struct, right?